Library/Homebrew/cmd/update.rb

```diff
@@ -216,6 +216,20 @@ def pull!(options = {})
       @initial_branch = ""
     end
 
+    # used for testing purpose, for example for testing formula migrations when
+    # some formula gets renamed on your branch and you test migrations by
+    # running 'brew update --from-branch"
+    if ARGV.include?("--from-branch")
+      @initial_revision = `git rev-parse -q --verify master`.chomp
```

Comment:
Maybe replace the hard-coded `master` with `#{@upstream_branch}` so it won't fail for the rare cases without a master branch (see #43865)? Same for the `odie` message below.